### ISSUE TITLE
string_lit_as_bytes should also detect `"str".to_string().into_bytes()`

### ISSUE THREAD

**Issue by repi** [rust-lang/rust-clippy#6927]
Lint name: `string_lit_as_bytes `

I tried this code:

```rust
let bytes1 = "Length".as_bytes();

let bytes2 = "Length".to_string().into_bytes();
```

I expected to see this happen:  for the second case (`bytes2`) to also be detected by the lint and recommend using `b"Length" instead

Instead, this happened: only the first case for `bytes1` was detected.

This came from real user code we had and manually fixed, would be nice if the lint can be extended to this case as well.

### Meta

- clippy 0.0.212 (cb75ad5d 2021-02-10)
- `rustc -Vv`:
  ```
   rustc 1.50.0 (cb75ad5db 2021-02-10)
   binary: rustc
   commit-hash: cb75ad5db02783e8b0222fee363c5f63f7e2cf5b
   commit-date: 2021-02-10
   host: x86_64-pc-windows-msvc
   release: 1.50.0
  ```

**Comment by ebobrow** [comment]
I'd like to give this a try.

@rustbot claim
